node/state_api/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+    let miner_actor_state: State =
+        state_manager.load_actor_state(&actor, &tipset.parent_state())?;
+    Ok(compute_proving_period_deadline(
+        miner_actor_state.proving_period_start,
+        tipset.epoch(),
+    ))
+}
+
+/// returns a single non-expired Faults that occur within lookback epochs of the given tipset
+pub fn state_miner_faults<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<BitField, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::get_miner_faults(&state_manager, &tipset, actor).map_err(|e| e.into())
+}
+
+/// returns all non-expired Faults that occur within lookback epochs of the given tipset
+pub fn state_all_miner_faults<DB>(
+    state_manager: &StateManager<DB>,
+    look_back: ChainEpoch,
+    end_tsk: &TipsetKeys,
+) -> Result<Vec<Fault>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(end_tsk)?;
+    let cut_off = tipset.epoch() - look_back;
+    let miners = state_manager::utils::list_miner_actors(&state_manager, &tipset)?;
+    let mut all_faults = Vec::new();
+    miners
+        .iter()
+        .map(|m| {
+            let miner_actor_state: State = state_manager
+                .load_actor_state(&m, &tipset.parent_state())
+                .map_err(|e| e.to_string())?;
+            let block_store = state_manager.get_block_store_ref();
+            miner_actor_state.for_each_fault_epoch(
+                block_store,
+                |fault_start: u64, _| -> Result<(), String> {
+                    if fault_start >= cut_off {
+                        all_faults.push(Fault {
+                            miner: *m,
+                            fault: fault_start,
+                        })
+                    }
+                    Ok(())
+                },
+            )
+        })
+        .collect::<Result<Vec<_>, String>>()?;
+    Ok(all_faults)
+}
+/// returns a bitfield indicating the recovering sectors of the given miner
+pub fn state_miner_recoveries<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<BitField, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::get_miner_recoveries(&state_manager, &tipset, actor).map_err(|e| e.into())
+}
+
+/// returns the power of the indicated miner
+pub fn state_miner_power<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<(Option<Claim>, Claim), BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::get_power(&state_manager, &tipset, Some(actor)).map_err(|e| e.into())
+}
+
+pub fn state_page_collateral<DB>(
```

Comment:
```suggestion
pub fn state_pledge_collateral<DB>(
```
Assume this is what you mean, can you add a TODO on this function also, so this doesn't stay unimplemented later please?